src/librustc_resolve/lib.rs

```diff
@@ -263,13 +263,16 @@ fn resolve_struct_error<'sess, 'a>(resolver: &'sess Resolver,
             err.span_label(span, "used in a pattern more than once");
             err
         }
-        ResolutionError::UndeclaredLabel(name) => {
+        ResolutionError::UndeclaredLabel(name, lev_candidate) => {
             let mut err = struct_span_err!(resolver.session,
                                            span,
                                            E0426,
                                            "use of undeclared label `{}`",
                                            name);
             err.span_label(span, format!("undeclared label `{}`", name));
```

Comment:
Could you replace the default label with Levenshtein-based message instead to avoid multiple messages?
```
if some lev_candidate {
    span_label("did you mean...")
} else {
    span_label("undeclared label")
}
```

Reply:
Sure!